File: BNSIosObjC/BNSIosObjC/ViewController.m
//
//  ViewController.m
//  TestObjCPebbleConnectivity
//
//  Created by Kristina Gancheva on 11/11/14.
//  Copyright (c) 2014 Peperzaken. All rights reserved.
//

#import <QuartzCore/QuartzCore.h>
#import "ViewController.h"
#import "PebbleDelegate.h"
#import "UIColorExtension.h"

@interface ViewController ()
@end

@implementation ViewController {
    __weak IBOutlet UITextView *commTracker;
    __weak IBOutlet UIButton *notifyButton;
}

- (void)viewDidLoad {
    commTracker.layer.borderColor = [[UIColor colorFromHex: @"#33b5e5"] CGColor];
    commTracker.layer.borderWidth = 3.0f;
    commTracker.autoresizingMask = UIViewAutoresizingFlexibleHeight;
    
    notifyButton.backgroundColor = [UIColor colorFromHex:@"#99cc00"];
    [notifyButton drawRect: CGRectMake(0, 0, 200, 80)];
    notifyButton.layer.cornerRadius = 10.0;
    [notifyButton setTitleColor: [UIColor whiteColor] forState: UIControlStateNormal];
    [notifyButton setTitle:@"Notify Watch" forState:UIControlStateNormal];
    
    [[PebbleDelegate sharedInstance] addObserver:self forKeyPath:@"notification" options: NSKeyValueObservingOptionNew context:nil];
    
    [super viewDidLoad];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(NSDictionary *)change context:(void *)context
{
    if ([keyPath isEqual:@"notification"]) {
        [self updatePhone:[PebbleDelegate sharedInstance].notification];
    } else {
        [super observeValueForKeyPath:keyPath ofObject:object change:change context:context];
    }
}

- (IBAction)updateWatch:(id)sender {
    [[PebbleDelegate sharedInstance] updateWatch];
}

- (void)updatePhone:(NSString *)update {
    //Update commTracker
    commTracker.text = [commTracker.text stringByAppendingString:update];
    commTracker.text = [commTracker.text stringByAppendingString:@"\n"];
}

- (void)clearCommTracker {
    commTracker.text = @"";
}

@end

File: BNSIosSwift/BNSIos_2/ViewController.swift
//
//  ViewController.swift
//  BNSIos_2
//
//  Created by Kristina Gancheva on 27/10/14.
//  Copyright (c) 2014 Peperzaken. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet weak var commTracker: UITextView!
    @IBOutlet weak var notifyButton: UIButton!
    
    //MARK: - ViewController-Lifecycle
    override func viewDidLoad() {
        commTracker.layer.borderColor = UIColor(rgba: "#33b5e5").CGColor
        commTracker.layer.borderWidth = 3.0
        commTracker.autoresizingMask = UIViewAutoresizing.FlexibleHeight
        
        notifyButton.backgroundColor = UIColor(rgba: "#99cc00")
        notifyButton.drawRect(CGRectMake(0, 0, 200, 80))
        notifyButton.layer.cornerRadius = 10.0
        notifyButton.setTitleColor(UIColor.whiteColor(), forState: UIControlState.Normal)
        notifyButton.setTitle("Notify Watch", forState: UIControlState.Normal)
        
        // Using key value observing for detecting changes in receivedMessage from Pebble and changes in currentLocation.
        PebbleDelegate.sharedInstance.addObserver(self, forKeyPath: "notification", options: NSKeyValueObservingOptions.New, context: nil)
        
        super.viewDidLoad()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    override func observeValueForKeyPath(keyPath: String, ofObject object: AnyObject, change: [NSObject : AnyObject], context: UnsafeMutablePointer<Void>) {
        if(keyPath == "notification") {
            updatePhone(PebbleDelegate.sharedInstance.notification)
        }
    }
    
    @IBAction func updateWatch(sender: AnyObject) {
        PebbleDelegate.sharedInstance.updateWatch()
    }
    
    //MARK: - UITextView
    func updatePhone(update: String) {
        commTracker.text = commTracker.text + update
        commTracker.text = commTracker.text + "\n"
    }
    
    func clearCommTracker() {
        commTracker.text = ""
    }
    
}

